test/end-to-end/offer.js

```diff
@@ -1103,4 +1109,400 @@ describe("Offer endpoint tests", () => {
             });
         });
     });
+
+    describe("POST /offers/edit/:/offerId", () => {
+        let createOffer,
+            expired_test_offer,
+            grace_period_over_test_offer,
+            grace_period_valid_test_offer,
+            future_test_offer,
+            clean_test_offer;
+
+        beforeAll(async () => {
+            await Offer.deleteMany({});
+            await test_agent.del("/auth/login");
+            createOffer = async (offer) => {
+                const { _id, owner, ownerName, jobMinDuration, jobMaxDuration } = await Offer.create({
+                    ...offer,
+                    owner: test_company._id.toString(),
+                    ownerName: test_company.name,
+                    jobMinDuration: 4,
+                    jobMaxDuration: 5,
+                });
+                return {
+                    ...offer,
+                    owner: owner.toString(),
+                    ownerName,
+                    _id: _id.toString(),
+                    jobMinDuration,
+                    jobMaxDuration
+                };
+            };
```

Comment:
It feels like you should modify the `generateTestOffer` to allow any parameter customization. Something like:

```js
const generateTestOffer = ({...params}) => ({
title: <deafultTitle>
otherparam: <defaultvalue>
...params // this will override the  default params with the passed in values inside the object argument
})
```